src/main/java/net/neoforged/neoforge/transfer/handlers/wrappers/CombinedResourceHandler.java

```diff
@@ -13,7 +13,7 @@
 /**
  * A resource handler that wraps multiple resource handlers, concatenating all their indices into one large handler.
  * <p>
- * <strong>This wrapper assumes that all internal handlers have a constant size.</strong>
+ * <strong>For this handler to work correctly, wrapped handlers must have a constant {@link IResourceHandler#size()}.</strong>
```

Comment:
Not quite. It will also work fine with "dynamic" handlers by always using the size at creation time (so if the handler expands, further indices will not be visible. And if it shrinks it's expected to be lenient anyway and allow no-op insertion into the removed slots). Maybe we could incorporate that information in the javadoc?